### PR TITLE
Add random ID, for the devices instead of the address.

### PR DESCRIPTION
**Pull request by fokinv**

### CODE REVIEW

**Comment by mmatyas** [comment]
good to me, it looks

File: components/net/bluetooth_thread.rs [hunk 1]
```diff
@@ -203,6 +212,20 @@ impl BluetoothManager {
         None
     }
 
+    #[allow(unused_assignments)]
+    fn generate_device_id(&mut self) -> String {
+        let mut device_id = String::new();
+        let mut rng = rand::thread_rng();
+        loop {
+            device_id = rng.gen_range(u32::min_value(), u32::max_value()).to_string();
```
Comment (dati91) [inline components/net/bluetooth_thread.rs]: `device_id = rng.gen::<u32>().to_string()` would be shorter

File: components/net/bluetooth_thread.rs [hunk 2]
```diff
@@ -417,14 +437,18 @@ impl BluetoothManager {
...
         if let Some(address) = self.select_device(matched_devices) {
-            if let Some(device) = self.get_device(&mut adapter, address.as_str()) {
-                let message = Ok(BluetoothDeviceMsg {
-                                     id: address,
-                                     name: device.get_name().ok(),
-                                     appearance: device.get_appearance().ok(),
-                                     tx_power: device.get_tx_power().ok().map(|p| p as i8),
-                                     rssi: device.get_rssi().ok().map(|p| p as i8),
-                                 });
+            let new_id = match self.address_to_id.get(&address) {
```
Comment (dati91) [inline components/net/bluetooth_thread.rs]: We should call this `device_id` or something like that.

File: components/net/bluetooth_thread.rs [hunk 3]
```diff
@@ -417,14 +437,18 @@ impl BluetoothManager {
...
+                Some(id) => id.clone(),
+                None => return drop(sender.send(Err(String::from(DEVICE_MATCH_ERROR)))),
+            };
+            if let Some(device) = self.get_device(&mut adapter, &new_id) {
+                    let message = Ok(BluetoothDeviceMsg {
```
Comment (dati91) [inline components/net/bluetooth_thread.rs]: indent

**Comment by dati91** [comment]
After the changes you can squash the commits

File: components/net/bluetooth_thread.rs [hunk 4]
```diff
@@ -417,9 +437,13 @@ impl BluetoothManager {
                                                            .filter(|d| matches_filters(d, options.get_filters()))
                                                            .collect();
         if let Some(address) = self.select_device(matched_devices) {
-            if let Some(device) = self.get_device(&mut adapter, address.as_str()) {
+            let device_id = match self.address_to_id.get(&address) {
```
Comment (dati91) [inline components/net/bluetooth_thread.rs]: maybe this `match` should be another `if let` instead

**Comment by zakorgy** [comment]
lgtm

**Comment by dati91** [comment]
We should change the commit name.

**Pull request merged**